plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServiceDetailViews.java

```diff
@@ -94,10 +94,14 @@ public void onVisible(IObservableValue serviceObservable, DataBindingContext dbc
 
 			nameText.setText(service.getName());
 			namespaceText.setText(service.getNamespace());
-			labelsText.setText(
-					org.jboss.tools.openshift.common.core.utils.StringUtils.toString(service.getLabels()));
-			selectorsText.setText(
-					org.jboss.tools.openshift.common.core.utils.StringUtils.toString(service.getSelector()));
+			String labels = org.jboss.tools.openshift.common.core.utils.StringUtils.toString(service.getLabels());
+			if(labels != null) {
+				labelsText.setText(labels);
+			}
```

Comment:
what if value is null? the text seems not to get reseted and would then still display the old value? shouldnt we reset it in this case?
Btw. org.apache.commons.lang.StringUtils.defaultString(string) returns an empty string if the input is null.